db_manager.py

```diff
@@ -37,63 +37,53 @@ def track_new_thread(self, url):
         """
             starts tracking a new thread by adding some info to the threads table. When a new thread is tracked, it automatically
             sets the values (last_page_num, last_post_num) to (1,1).
-            Returns True if inserted, False if it encounters a conflict
+            Returns the thread_id
         """
         cursor = self.db_connection.cursor()
+        sql_command = "INSERT INTO threads (url, active, last_page_num, last_post_num) VALUES (?, ?, ?, ?)"
         try:
-            cursor.execute('INSERT INTO threads (url, active, last_page_num, last_post_num) VALUES (?, ?, ?, ?)', (url, 1, 1, 1))
+            cursor.execute(sql_command, (url, 1, 1, 1))
         except sqlite3.IntegrityError:
-            print("already tracked") #NOTE: some better kind of return here?
-            return False
+            if display : print("already tracked")
         else:
             self.db_connection.commit()
+        finally:
+            sql_command = "SELECT thread_id FROM threads WHERE url=?"
+            cursor.execute(sql_command, (url,))
+            thread_id = cursor.fetchone()[0]
+            cursor.close()
+            return thread_id
 
-        cursor.close()
-        return True
-
-    def insert_raw_post(self, raw_data, thread_id, force=False):
+    def insert_forum_post(self, forum_posts, force=False):
         """
-        if force is passed as True, then raw_posts will overwrite any conflicts
-        raw_data should be a list of tuples in format (timestamp, username, post_body)
-        will insert raw post data into the raw_posts table. We should aim to deprecate this
-        Returns True if inserted, False if it encounters a conflict
+        if force is passed as True, then this function will overwrite any database conflicts
+        if display is passed as True, then this function will output information to the terminal
```

Comment:
this comment is outdated and needs deleting. display is now a class function, not an argument